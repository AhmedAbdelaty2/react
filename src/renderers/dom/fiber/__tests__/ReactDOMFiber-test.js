/**
 * Copyright 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails react-core
 */

'use strict';

var React = require('React');
var ReactDOMFiber = require('ReactDOMFiber');

describe('ReactDOMFiber', () => {
  var containerEl;

  beforeEach(() => {
    // to supress a warning that ReactDOMFiber is an experimental renderer.
    spyOn(console, 'error');
    containerEl = document.createElement('div');
  });

  it('should be able to render string as children', () => {
    var Foo = ({child}) => <div>{child}</div>;
    ReactDOMFiber.render(
      <Foo child="test" />,
      containerEl
    );
    expect(containerEl.textContent).toBe('test');
  });

  it('should be able to render number as children', () => {
    var Foo = ({child}) => <div>{child}</div>;
    ReactDOMFiber.render(
      <Foo child={1} />,
      containerEl
    );
    expect(containerEl.textContent).toBe('1');
  });
});
