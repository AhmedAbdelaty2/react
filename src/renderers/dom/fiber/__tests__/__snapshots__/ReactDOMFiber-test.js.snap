// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactDOMFiber should warn when replacing a container which was manually updated outside of React 1`] = `""`;
